src/ol/overlay.js

```diff
@@ -455,27 +455,48 @@ ol.Overlay.prototype.setPositioning = function(positioning) {
 
 
 /**
- * @private
+ * Modify the visibility of the element.
+ * @param {boolean} visible
+ * @protected
  */
-ol.Overlay.prototype.updatePixelPosition_ = function() {
+ol.Overlay.prototype.setVisibility = function(visible) {
+  if (this.rendered_.visible ^ visible) {
```

Comment:
Since we're not using bitwise operators for this type of comparison elsewhere, I think `!==` would be clearer here.